src/cli.rs

```diff
@@ -0,0 +1,128 @@
+//! Contains structures and functionality for the binary
+use std::path::PathBuf;
+
+use nemo::{error::Error, io::OutputFileManager};
+
+const DEFAULT_OUTPUT_DIRECTORY: &str = "results";
+
+/// Cli Arguments related to logging
+#[derive(clap::Args, Debug)]
+pub struct LoggingArgs {
+    /// Sets the verbosity of logging if the flags -v and -q are not used
+    #[arg(long = "log", value_parser=clap::builder::PossibleValuesParser::new(["error", "warn", "info", "debug", "trace"]), group = "verbosity")]
+    log_level: Option<String>,
+    /// Sets log verbosity (multiple times means more verbose)
+    #[arg(short, long, action = clap::builder::ArgAction::Count, group = "verbosity")]
+    verbose: u8,
+    /// Sets log verbosity to only log errors
+    #[arg(short, long, group = "verbosity")]
+    quiet: bool,
+}
+
+impl LoggingArgs {
+    /// Initialising Logging
+    ///
+    /// Sets the logging verbosity to the given log-level in the following order:
+    ///  * `Info`, `Debug`, `Trace`; depending on the count of `-v`
+    ///  * `Error` when `-q` is used
+    ///  * The `NMO_LOG` environment variable value
+    ///  * `Warn` otherwise
+    pub fn initialize_logging(&self) {
+        let mut builder = env_logger::Builder::new();
+        builder.parse_env("NMO_LOG");
```

Comment:
Yeah, reading from the environment was broken previously, because the cli arguments always overwrote it (even if there were none). While fixing this I also changed the name, just because the environment argument does not control anything rust specific but can be seen as an extension of nemo's cli interface.

Reply:
I can take care of the scripts or revert, but would probably prefer `NMO_LOG`

Reply:
I think it was actually working as intended – certainly working as documented, though. In general, I would like to be able to override the environment setting with the command line arguments. Regarding “anything rust specific”, it certainly _does_ also control logging in crates that are not our own (but we might still want to rename it).

Reply:
> it certainly does also control logging in crates that are not our own

ah, I was not aware of that.

Previously when self.verbose was zero (i.e. no verbose flag) and no other cli parameters were set, then the log filter would be set to `Warn`, hence why I added the if > 0.

That was also the reason for me to add the trace parameter in the blackbox tests, which is now removed, because it slowed down the tests significantly.

Reply:
> Previously when self.verbose was zero (i.e. no verbose flag) and no other cli parameters were set, then the log filter would be set to Warn, hence why I added the if > 0.

I agree, that should not happen.